src/service/AssignmentService.ts

```
@@ -18,27 +20,66 @@ const question = new Question(
     `SELECT * FROM Employees`
 );
 
-function compareResults(a: Array<any>, b: Array<any>) {
-    if (a.length !== b.length) return false;
+function compareResults(a: QueryResult, b: QueryResult) {
+    if (a.rows.length !== b.rows.length) return false;
 
     return JSON.stringify(a) === JSON.stringify(b);
 }
 
+function formatQueryResult(result: QueryResult) {
+    return result.success ? result.rows : { "error": result.error };
+}
+function formatSingleQueryResult(result: QueryResult) {
+    return result.success
+        ? (result.rows.length > 0 ? result.rows[0] : { "error": "No data was found" })
+        : { "error": result.error };
```

It's not necessarily an error that we found no data, it could just be that there are no questions for a given assignment. It's probably okay to return a success with no data and no error message.

Incidentally if you did want to throw an error, we should also be giving back error status codes too. If we failed to connect to or hit our database I think we'd want a 500 for that. I'm guessing that right now if you were to get the error response in postman it would still tell you the query status is successful with some kind of 200.